src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +174,313 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  EXPECT_CALL(*mock_app_, policy_app_id()).Times(0);
+
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_)).WillOnce(Return(true));
+
+  request_->Run();
+}
+
+TEST_F(SetAppIconRequestTest,
+       CopyToIconStorage_PolicyAppIdEqualFolderName_CantCreateIcon) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(mock_app_));
+
+  MockProtocolHandler mock_protocol_handler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_policy_id";
+  EXPECT_CALL(*mock_app_, policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_)).WillOnce(Return(true));
+
+  file_system::DeleteFile(kIconFolder + "/" + app_policy_id);
+  file_system::CreateDirectory(kIconFolder + "/" + app_policy_id);
+  EXPECT_FALSE(file_system::CreateFile(kIconFolder + "/" + app_policy_id));
+  request_->Run();
+}
+
+TEST_F(SetAppIconRequestTest,
+       RemoveOldestIcons_AllCorrectData_OldIconDeletedSuccessfully) {
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  MockProtocolHandler mock_protocol_handler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillOnce(Return(max_supported_protocol_version));
+
+  const uint32_t max_storage_size_before_removal = 11u;
+  const uint32_t max_storage_size_after_removal = 120u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillOnce(ReturnRef(max_storage_size_before_removal))
```

Comment:
@BSolonenko Times(3).WillRepeatedly(Return....)..

Reply:
@AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/b69062aa4d5eeea9c38a0ac0745d1afe7bfce283